### ISSUE TITLE
Missing block causes error when publishing event

### ISSUE THREAD

**Issue by pascalbetz** [krisleech/wisper#145]
When the block is omitted for ```on``` then a ```broadcast``` causes an __undefined method `call' for nil:NilClass (NoMethodError)__ error.

I'd be nice to fail fast and show an appropriate error message. What do you think?

Following shows the problem:

```ruby
require 'wisper'

class Thing
  include Wisper::Publisher
  def do_something
    publish(:something, self)
  end
end

thing = Thing.new

thing.on(:something)
thing.do_something


somewhere/ruby-2.2.0/gems/wisper-1.6.1/lib/wisper/registration/block.rb:5:in `broadcast': undefined method `call' for nil:NilClass (NoMethodError)
	from somewhere/ruby-2.2.0/gems/wisper-1.6.1/lib/wisper/publisher.rb:65:in `block in broadcast'
	from somewhere/ruby-2.2.0/lib/ruby/2.2.0/set.rb:283:in `each_key'
	from somewhere/ruby-2.2.0/lib/ruby/2.2.0/set.rb:283:in `each'
	from somewhere/ruby-2.2.0/gems/wisper-1.6.1/lib/wisper/publisher.rb:64:in `broadcast'
	from somewhere/snippets/broadcast.rb:6:in `do_something'
	from somewhere/snippets/broadcast.rb:16:in `<main>'```
```

(I hit this problem when I first used ```rectify``` and accidentally left out the block)

**Comment by krisleech** [comment]
Totally happy with this :) I just need to fix Travis so JRuby uses Ruby 2.0 syntax - I think we need JRuby 9000.

**Comment by pascalbetz** [comment]
I saw other PRs with failing jruby builds so I figured it must have been something like that. 
Let me know if I need to rebase or update.
